src/util/positioning.ts

```diff
@@ -1,167 +1,49 @@
   }
-
-  /*
-    Return false if the element to position is outside the viewport
-  */
-  positionElements(hostElement: HTMLElement, targetElement: HTMLElement, placement: string, appendToBody?: boolean):
-      boolean {
-    const[placementPrimary = 'top', placementSecondary = 'center'] = placement.split('-');
-
-    const hostElPosition = appendToBody ? this.offset(hostElement, false) : this.position(hostElement, false);
-    const targetElStyles = this.getAllStyles(targetElement);
-
-    const marginTop = parseFloat(targetElStyles.marginTop);
-    const marginBottom = parseFloat(targetElStyles.marginBottom);
-    const marginLeft = parseFloat(targetElStyles.marginLeft);
-    const marginRight = parseFloat(targetElStyles.marginRight);
-
-    let topPosition = 0;
-    let leftPosition = 0;
-
-    switch (placementPrimary) {
-      case 'top':
-        topPosition = (hostElPosition.top - (targetElement.offsetHeight + marginTop + marginBottom));
-        break;
-      case 'bottom':
-        topPosition = (hostElPosition.top + hostElPosition.height);
-        break;
-      case 'start':
-        leftPosition = (hostElPosition.left - (targetElement.offsetWidth + marginLeft + marginRight));
-        break;
-      case 'end':
-        leftPosition = (hostElPosition.left + hostElPosition.width);
-        break;
-    }
-
-    switch (placementSecondary) {
-      case 'top':
-        topPosition = hostElPosition.top;
-        break;
-      case 'bottom':
-        topPosition = hostElPosition.top + hostElPosition.height - targetElement.offsetHeight;
-        break;
-      case 'start':
-        leftPosition = hostElPosition.left;
-        break;
-      case 'end':
-        leftPosition = hostElPosition.left + hostElPosition.width - targetElement.offsetWidth;
-        break;
-      case 'center':
-        if (placementPrimary === 'top' || placementPrimary === 'bottom') {
-          leftPosition = (hostElPosition.left + hostElPosition.width / 2 - targetElement.offsetWidth / 2);
-        } else {
-          topPosition = (hostElPosition.top + hostElPosition.height / 2 - targetElement.offsetHeight / 2);
-        }
-        break;
-    }
-
-    /// The translate3d/gpu acceleration render a blurry text on chrome, the next line is commented until a browser fix
-    // targetElement.style.transform = `translate3d(${Math.round(leftPosition)}px, ${Math.floor(topPosition)}px, 0px)`;
-    targetElement.style.transform = `translate(${Math.round(leftPosition)}px, ${Math.round(topPosition)}px)`;
-
-    // Check if the targetElement is inside the viewport
-    const targetElBCR = targetElement.getBoundingClientRect();
-    const html = document.documentElement;
-    const windowHeight = window.innerHeight || html.clientHeight;
-    const windowWidth = window.innerWidth || html.clientWidth;
-
-    return targetElBCR.left >= 0 && targetElBCR.top >= 0 && targetElBCR.right <= windowWidth &&
-        targetElBCR.bottom <= windowHeight;
+  if (baseClass) {
```

Comment:
Like can we have `baseClass` that is not set really? Once again unit tests would answer all these questions...